base/src/commonTest/kotlin/flow/HotFlowableTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.splendo.kaluga.base.test.flow
+
+import com.splendo.kaluga.base.flow.HotFlowable
+import com.splendo.kaluga.base.runBlocking
+import com.splendo.kaluga.test.BaseTest
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.channels.BroadcastChannel
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.take
+import kotlinx.coroutines.launch
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class HotFlowableTest : BaseTest() {
+
+    lateinit var flowable: HotFlowable<Int>
+    lateinit var broadcastChannel: BroadcastChannel<Int>
+
+    @BeforeTest
+    fun setUp() {
+        super.beforeTest()
+        broadcastChannel = ConflatedBroadcastChannel()
+        flowable = HotFlowable(0) {broadcastChannel}
+    }
+
+    @Test
+    fun `Test Flowing`() = runBlocking {
+        assertEquals(0, broadcastChannel.asFlow().first())
+        flowable.set(1)
+        assertEquals(1, broadcastChannel.asFlow().first())
+        assertEquals(1, flowable.flow().first())
+    }
+
+    @Test
+    fun `Test Multiple Flowing`() = runBlocking {
+        val values1 = CompletableDeferred<List<Int>>()
+        val values2 = CompletableDeferred<List<Int>>()
+        val scope = MainScope()
+        val job1 = scope.launch {
+            val values = emptyList<Int>().toMutableList()
+            flowable.flow().take(3).collect { value ->
+                values.add(value)
+                if (values.size == 3) {
+                    values1.complete(values)
+                }
+            }
+        }
+        scope.launch {
+            flowable.set(1)
```

Comment:
This is not guaranteed to run before the job2 below. It is if you use `setBlocking`, or I believe if we use `runBlockingTest` to run the test (we have to expose it in base though)